lib/src/model/clicker/click_handler.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:async';
+
+import 'package:link_five/src/logic/actions/move_tile_action.dart';
+import 'package:link_five/src/logic/actions/place_tile_action.dart';
+import 'package:link_five/src/logic/game_action.dart';
+import 'package:link_five/src/model/game/game_state.dart';
+import 'package:link_five/src/model/game/player_color.dart';
+import 'package:link_five/src/model/game/tile_location.dart';
+
+class ClickHandler {
+  TileLocation? _source;
+  TileLocation? get source => _source;
+  var _stateStreamController = StreamController<TileLocation?>.broadcast();
+  Stream<TileLocation?> get stateStream => _stateStreamController.stream;
+
+  GameAction? handleGameClick(
```

Comment:
This method has a lot of nesting. It would probably be more readable if broken up into some functions.